packages/app/src/debug/DebugContainer.vue

```diff
@@ -1,16 +1,30 @@
 <template>
-  <div>
+  <div class="h-full">
     <DebugError
       v-if="showError"
     />
     <div
-      v-else-if="loginConnectStore.user.isLoggedIn && loginConnectStore.project.isProjectConnected && run"
+      v-else-if="loginConnectStore.user.isLoggedIn && loginConnectStore.project.isProjectConnected && run && run.status"
+      class="flex flex-col h-full"
     >
       <DebugPageHeader
         :gql="run"
         :commits-ahead="0"
       />
+      <DebugPageDetails
```

Comment:
Should `DebugPageDetails` declare a `fragment` and just receive a single prop (`gql`)? It would save passing a bunch of individual properties, and you wouldn't need to deal with the `:? null | undefined` issue around types.

Reply:
I don't have a firm stance on this but I'm used to being explicit about what we're passing down to the component. This would solve the `null | undefined` issue but only for `DebugPageDetails`. Same issue would apply with all of the individual components that we render inside of `DebugPageDetails`. 

Not crazy about the idea of passing one giant `gql` prop down rather than breaking them out into what info we actually need for the component

Reply:
I also like decomposing `gql` into individual props, makes it easier to test (IMHO), but this is a pretty extensive set of props. Could some of these be consolidated, like `cancelledAt`, `cancelledByFullName`, and `cancelledByEmail` combined into a single `cancellation` object prop?